src/test/java/javaslang/collection/IntMap.java

```diff
@@ -121,9 +121,7 @@ private Object readResolve() {
 
     @Override
     public IntMap<Object> flatten() {
-        final Seq<Object> seq = original.values().flatten();
-        final Map<Integer, Object> map = seq.zipWithIndex().toMap(t -> Tuple.of(t._2, t._1));
-        return IntMap.of(map);
+        throw new UnsupportedOperationException();
```

Comment:
Yep. In flatten() tests I use emptyMap() constructor which provides Map directly, instead of empty() which provides IntMap wrapper